src/Driver/Pdo/Connection.php

```diff
@@ -0,0 +1,332 @@
+use Laminas\Db\Adapter\Driver\ResultInterface;
+use Laminas\Db\Adapter\Driver\StatementInterface;
+use Laminas\Db\Adapter\Exception;
+use Laminas\Db\Adapter\Exception\RunTimeException;
+use PDO;
+use PDOException;
+use PDOStatement;
+
+use function array_diff_key;
+use function is_array;
+use function is_int;
+use function str_replace;
+use function str_starts_with;
+use function strtolower;
+use function substr;
+
+class Connection extends AbstractPdoConnection implements ConnectionInterface, PdoDriverAwareInterface
+{
+    protected PdoDriverInterface $driver;
+
+    /** @var PDO */
+    protected $resource;
+
+    protected ?string $dsn;
+
+    /**
+     * Constructor
+     *
+     * @throws Exception\InvalidArgumentException
+     */
+    public function __construct(PDO|array|null $connectionParameters = null)
+    {
+        parent::__construct($connectionParameters);
+
+        if (is_array($connectionParameters)) {
+            $this->setConnectionParameters($connectionParameters);
+        } elseif ($connectionParameters instanceof PDO) {
+            $this->setResource($connectionParameters);
+        } elseif (null !== $connectionParameters) {
+            throw new Exception\InvalidArgumentException(
+                '$connection must be an array of parameters, a PDO object or null'
+            );
+        }
+    }
+
+    /**
+     * Set driver
+     *
+     * @return $this Provides a fluent interface
+     */
+    public function setDriver(PdoDriverInterface $driver): static
+    {
+        $this->driver = $driver;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getCurrentSchema(): bool|string
+    {
+        if (! $this->isConnected()) {
+            $this->connect();
+        }
+
+        /** @var PDOStatement $result */
+        $result = $this->resource->query('PRAGMA database_list');
+        if ($result instanceof PDOStatement) {
+            return $result->fetchColumn();
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws Exception\InvalidConnectionParametersException
+     * @throws Exception\RuntimeException
+     */
+    public function connect(): static
+    {
+        if ($this->resource) {
+            return $this;
+        }
+
+        $dsn     = $username = $password = $hostname = $database = null;
+        $options = [];
+        foreach ($this->connectionParameters as $key => $value) {
+            switch (strtolower($key)) {
+                case 'dsn':
+                    $dsn = $value;
+                    break;
+                case 'driver':
+                    $value = strtolower((string) $value);
+                    if (str_starts_with($value, 'pdo')) {
+                        $pdoDriver = str_replace(['-', '_', ' '], '', $value);
+                        $pdoDriver = substr($pdoDriver, 3) ?: '';
+                    }
+                    break;
+                case 'pdodriver':
```

Comment:
I missed this in my initial review. I think it would serve us well to remove the one off parameters in favor of using just the 'driver' key. One of my goals is to standardize this to ease documenting all of the changes. I can not confirm this but my hunch is that we find these one offs due to so many different contributors providing code over the years.